src/hardware/disk_noise.h

```diff
@@ -0,0 +1,89 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2025-2025  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_DISK_NOISE_H
+#define DOSBOX_DISK_NOISE_H
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "mixer.h"
+
+class DiskNoiseDevice {
+public:
+	DiskNoiseDevice(const DiskType disk_type,
+					const bool enable_disk_noise,
+	                const std::string& spin_up_sample_path,
+	                const std::string& spin_sample_path,
+	                const std::vector<std::string>& seek_sample_paths,
+	                bool loop_spin_sample);
+	~DiskNoiseDevice();
+	void ActivateSpin();
+	void PlaySeek();
+	AudioFrame GetNextFrame();
+
+private:
+	bool enable_disk_noise    = false;
+
+	struct SpinSample {
+		std::vector<float> spin_up_sample;
+		std::vector<float> sample;
+		bool loop = false;
+		std::vector<float>::iterator spin_up_it = spin_up_sample.end();
```

Comment:
This doesn't make too much sense, just initialise the iterators with `{}`. You should initialise the iterator properly in the implementation after the buffers have been filled to point at the start of the buffers.

Reply:
Also, you could use constant iterators eveywhere because you're only reading the data via iterators. That would be preferable.

https://www.geeksforgeeks.org/const-vs-regular-iterators-in-c-with-examples/